layer/profiles.cpp

```diff
@@ -3688,6 +3917,153 @@ bool JsonLoader::GetFormat(const Json::Value &formats, const std::string &format
     return valid;
 }
 
+bool QueueFamilyMatch(const VkQueueFamilyProperties &device_qfp, const VkQueueFamilyProperties &profile_qfp) {
+    if ((device_qfp.queueFlags & profile_qfp.queueFlags) != profile_qfp.queueFlags) {
+        return false;
+    }
+    if (device_qfp.queueCount < profile_qfp.queueCount) {
+        return false;
+    }
+    if (device_qfp.timestampValidBits != profile_qfp.timestampValidBits) {
+        return false;
+    }
+    if (device_qfp.minImageTransferGranularity.width >= profile_qfp.minImageTransferGranularity.width ||
+        device_qfp.minImageTransferGranularity.height >= profile_qfp.minImageTransferGranularity.height ||
+        device_qfp.minImageTransferGranularity.depth >= profile_qfp.minImageTransferGranularity.depth) {
+        return false;
+    }
+    return true;
+}
+
+bool JsonLoader::GetQueueFamilyProperties(const Json::Value &qf_props, ArrayOfVkQueueFamilyProperties *dest,
+                                          VkQueueFamilyGlobalPriorityPropertiesKHR *gp_dest,
+                                          VkVideoQueueFamilyProperties2KHR *vqfp_dest,
+                                          VkQueueFamilyCheckpointPropertiesNV *qfcp_dest,
+                                          VkQueueFamilyCheckpointProperties2NV *qfcp2_dest,
+                                          VkQueueFamilyQueryResultStatusProperties2KHR *qfqrsp_dest) {
```

Comment:
Maybe I'm missing something, but why the `VkQueueFamilyProperties` parameter is an array while the others aren't?
Each should be, as each queue family can have different global priorities, checkpoint properties, etc.

Reply:
I was told there can only be one of each in the pNext chain, duplicates are considered an error.

Reply:
Well, that's not what the spec says, also, the schema allows defining separate structures for each queue family definition index (it always did, even before it was generated).

What you probably refer to is that a single `pNext` chain can contain a structure only once. However, we're talking about multiple `pNext` chains here, one per queue family, each can have its own version of any of the chained structures.